tests/test_alerter.py

```diff
@@ -414,172 +419,165 @@ def test_alert_alerter_not_urgent(self):
 
 class TestMessageBuilding(unittest.TestCase):
     def setUp(self):
-        self.test_alerter = alerter.Alerter()
-        self.freeze_time_value = "2020-03-10 09:00"
-        self.tz = os.environ.get("TZ", "local")
-        self.expected_time_string = (
-            arrow.get("2020-03-10 09:00:00+00:00").to(self.tz).format()
-        )
-        a = arrow.now()
-        self.utcoffset = a.utcoffset()
+        # Force these ones to UTC to simplify; see also TestMessageBuildingTZ class
+        # Note: For Reasons, the decorator doesn't work well with this.
+        with mock.patch.dict(os.environ, {"TZ": "UTC"}, clear=True):
```

Comment:
With this one only, I could induce failures by doing, e.g.,
```
$ TZ=EST5EDT  poetry run pytest -vvv
```

somehow, `simplemonitor/Alerters/alerter.py` still would be reading `os.env.TZ` from the system in this case. I first tried seeing if I needed to use `importlib` to reload the libraries being tested, because maybe they were evaluating `os.getenv()` first, or overriding the decorator for `setup`, or that it was something related to using unittest test cases within pytest (which I'm guessing is it?); in the end, using a `with` in the setup routine here seemed to resolve it.

https://adamj.eu/tech/2020/10/13/how-to-mock-environment-variables-with-pythons-unittest/
https://adamj.eu/tech/2020/10/13/how-to-mock-environment-variables-with-pytest/